packages/valory/skills/staking_abci/behaviours.py

```diff
@@ -110,6 +105,12 @@ def service_staking_state(self) -> StakingState:
     @service_staking_state.setter
     def service_staking_state(self, state: StakingState) -> None:
         """Set the service's staking state."""
+
+        # The class StakingState is redefined in several packages.
+        # This conversion is required to use a single representation.
+        if not isinstance(state, StakingState):
+            state = StakingState(state.value)
+
```

Comment:
Review this approach

Reply:
@Adamantios please validate this approach

Reply:
Can you elaborate? Why is this necessary? It is only specified in the contract from where you imported it.

Reply:
`service_staking_state` is initialized with `StakingState` defined in 

https://github.com/valory-xyz/trader/blob/main/packages/valory/contracts/service_staking_token/contract.py#L30

and 

https://github.com/valory-xyz/trader/blob/feat/new-staking/packages/valory/contracts/staking_token/contract.py#L30

deppending of which of the 2 contracts is being used. These are 2 different classes and when service_staking_state  is compared, it will fail if it's not the class defined in this file. For this reason, this patch here ensures that it always uses StakingState imported in this file.

Reply:
So I think the best approach here would be to revert back to https://github.com/valory-xyz/trader/pull/209#discussion_r1465226383.

Reply:
This branch has already been tested.
I added an issue to ammend on the base branch before merging to main. https://github.com/valory-xyz/trader/issues/280